File: ios/NewsFeed/Core/Models/Article.swift
import Foundation

struct Article: Identifiable, Decodable, Hashable {
    let id: Int
    let title: String
    let slug: String?
    let excerpt: String?
    let content: String?
    let imageURL: String?
    let sourceURL: String?
    let publishedAt: String?
    let viewCount: Int
    let comments: Int
    let isBreaking: Bool
    let isFeatured: Bool
    let category: ArticleCategory
    let source: ArticleSource

    enum CodingKeys: String, CodingKey {
        case id, title, slug, excerpt, content, comments
        case imageURL = "image_url"
        case sourceURL = "source_url"
        case publishedAt = "published_at"
        case viewCount = "view_count"
        case isBreaking = "is_breaking"
        case isFeatured = "is_featured"
        case category, source
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id = try c.decode(Int.self, forKey: .id)
        title = try c.decodeIfPresent(String.self, forKey: .title) ?? ""
        slug = try c.decodeIfPresent(String.self, forKey: .slug)
        excerpt = try c.decodeIfPresent(String.self, forKey: .excerpt)
        content = try c.decodeIfPresent(String.self, forKey: .content)
        imageURL = try c.decodeIfPresent(String.self, forKey: .imageURL)
        sourceURL = try c.decodeIfPresent(String.self, forKey: .sourceURL)
        publishedAt = try c.decodeIfPresent(String.self, forKey: .publishedAt)
        viewCount = (try? c.decode(Int.self, forKey: .viewCount)) ?? 0
        comments = (try? c.decode(Int.self, forKey: .comments)) ?? 0
        isBreaking = (try? c.decode(Bool.self, forKey: .isBreaking)) ?? false
        isFeatured = (try? c.decode(Bool.self, forKey: .isFeatured)) ?? false
        category = (try? c.decode(ArticleCategory.self, forKey: .category)) ?? .empty
        source = (try? c.decode(ArticleSource.self, forKey: .source)) ?? .empty
    }
}

struct ArticleCategory: Decodable, Hashable {
    let id: Int?
    let name: String?
    let slug: String?
    let icon: String?
    let cssClass: String?

    enum CodingKeys: String, CodingKey { case id, name, slug, icon; case cssClass = "css_class" }

    static let empty = ArticleCategory(id: nil, name: nil, slug: nil, icon: nil, cssClass: nil)
}

struct ArticleSource: Decodable, Hashable {
    let id: Int?
    let name: String?
    let logoLetter: String?
    let logoColor: String?
    let logoBg: String?
    let url: String?

    enum CodingKeys: String, CodingKey {
        case id, name, url
        case logoLetter = "logo_letter"
        case logoColor = "logo_color"
        case logoBg = "logo_bg"
    }

    static let empty = ArticleSource(id: nil, name: nil, logoLetter: nil, logoColor: nil, logoBg: nil, url: nil)
}

File: api/v1/content/article.php
<?php
require_once __DIR__ . '/../_bootstrap.php';
require_once __DIR__ . '/../_articles_query.php';

api_method('GET');
api_rate_limit('content:article', 240, 60);

$id   = isset($_GET['id'])   ? (int)$_GET['id'] : 0;
$slug = isset($_GET['slug']) ? trim((string)$_GET['slug']) : '';

if (!$id && $slug === '') api_err('invalid_input', 'يلزم تمرير id أو slug', 422);

$db = getDB();
$where = $id ? 'a.id = ?' : 'a.slug = ?';
$param = $id ?: $slug;

$sql = articles_select_sql() . ", a.content
       FROM articles a
       LEFT JOIN categories c ON c.id = a.category_id
       LEFT JOIN sources    s ON s.id = a.source_id
       WHERE $where AND a.status = 'published' LIMIT 1";

// articles_select_sql() already starts with SELECT ... FROM articles a; rebuild precisely:
$sql = "SELECT
        a.id, a.title, a.slug, a.excerpt, a.content, a.image_url, a.source_url,
        a.is_breaking, a.is_featured, a.is_hero,
        a.view_count, a.comments, a.published_at, a.created_at,
        a.category_id, a.source_id,
        c.name AS category_name, c.slug AS category_slug, c.icon AS category_icon, c.css_class,
        s.name AS source_name, s.slug AS source_slug, s.logo_letter, s.logo_color, s.logo_bg, s.url AS source_site
      FROM articles a
      LEFT JOIN categories c ON c.id = a.category_id
      LEFT JOIN sources    s ON s.id = a.source_id
      WHERE $where AND a.status = 'published' LIMIT 1";

$st = $db->prepare($sql);
$st->execute([$param]);
$row = $st->fetch();
if (!$row) api_err('not_found', 'المقال غير موجود', 404);

// Increment view count (best-effort).
try {
    $db->prepare('UPDATE articles SET view_count = view_count + 1 WHERE id=?')->execute([(int)$row['id']]);
} catch (Throwable $e) {}

$article = api_format_article($row);
$article['content'] = $row['content'];

// Related articles by same category, excluding self.
$related = [];
if (!empty($row['category_id'])) {
    $rs = $db->prepare(articles_select_sql() . " WHERE a.status='published' AND a.category_id=? AND a.id<>? ORDER BY a.published_at DESC LIMIT 6");
    $rs->execute([(int)$row['category_id'], (int)$row['id']]);
    $related = array_map('api_format_article', $rs->fetchAll());
}

api_ok([
    'article' => $article,
    'related' => $related,
]);
